rewrite-groovy/src/main/java/org/openrewrite/groovy/CustomResolveVisitor.java

```diff
@@ -0,0 +1,1659 @@
+                    possibleOuterClassNodeMap.put(cce.getType(), possibleOuterClassNode);
+                }
+            }
+        }
+    }
+
+    private static String getDescription(final ClassNode node) {
+        return (node.isInterface() ? "interface" : "class") + " '" + node.getName() + "'";
+    }
+
+    protected Expression transformMethodCallExpression(final MethodCallExpression mce) {
+        Expression args = transform(mce.getArguments());
+        Expression method = transform(mce.getMethod());
+        Expression object = transform(mce.getObjectExpression());
+
+        resolveGenericsTypes(mce.getGenericsTypes());
+
+        MethodCallExpression ret = new MethodCallExpression(object, method, args);
+        ret.setGenericsTypes(mce.getGenericsTypes());
+        ret.setMethodTarget(mce.getMethodTarget());
+        ret.setImplicitThis(mce.isImplicitThis());
+        ret.setSpreadSafe(mce.isSpreadSafe());
+        ret.setSafe(mce.isSafe());
+        return ret;
+    }
+
+    protected Expression transformDeclarationExpression(final DeclarationExpression de) {
+        visitAnnotations(de);
+        Expression oldLeft = de.getLeftExpression();
+        checkingVariableTypeInDeclaration = true;
+        Expression left = transform(oldLeft);
+        checkingVariableTypeInDeclaration = false;
+        if (left instanceof ClassExpression) {
+            ClassExpression ce = (ClassExpression) left;
+            addError("you tried to assign a value to the class " + ce.getType().getName(), oldLeft);
+            return de;
+        }
+        Expression right = transform(de.getRightExpression());
+        if (right == de.getRightExpression()) {
+            fixDeclaringClass(de);
+            return de;
+        }
+        DeclarationExpression newDeclExpr = new DeclarationExpression(left, de.getOperation(), right);
+        newDeclExpr.setDeclaringClass(de.getDeclaringClass());
+        newDeclExpr.addAnnotations(de.getAnnotations());
+        newDeclExpr.copyNodeMetaData(de);
+        fixDeclaringClass(newDeclExpr);
+        return newDeclExpr;
+    }
+
+    // TODO: get normal resolving to set declaring class
+    private void fixDeclaringClass(final DeclarationExpression newDeclExpr) {
+        if (newDeclExpr.getDeclaringClass() == null && currentMethod != null) {
+            newDeclExpr.setDeclaringClass(currentMethod.getDeclaringClass());
+        }
+    }
+
+    protected Expression transformAnnotationConstantExpression(final AnnotationConstantExpression ace) {
+        AnnotationNode an = (AnnotationNode) ace.getValue();
+        ClassNode type = an.getClassNode();
+        resolveOrFail(type, " for annotation", an);
+        for (Map.Entry<String, Expression> member : an.getMembers().entrySet()) {
+            member.setValue(transform(member.getValue()));
+        }
+        return ace;
+    }
+
+    @Override
+    public void visitAnnotations(final AnnotatedNode node) {
```

Comment:
I think it's better to not copy the whole file, but just only override this method. So remove all the methods and only keep this method in tact. And then change:
```diff
- Expression adjusted = transformInlineConstants(newValue);
+ // OR: Disable inlining, as it causes the AST and source code to be out of sync.
+ // Expression adjusted = transformInlineConstants(newValue);
```

This way it's less error prone to changes to this file + way easier to spot what we changed in the first place.

Reply:
That would be nice. I tried that first, but there are a lot of private methods called here, which in turn call other private methods, etc. So in the end a lot of code ends up getting pulled in.

Reply:
I fear that it will end up being harder to sync if there are upstream changes. At the moment the file is a copy with just two lines removed.